core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -254,8 +254,16 @@ object SparkEnv extends Logging {
           if (port == 0 || rpcEnv.address == null) {
             port
           } else {
-            rpcEnv.address.port + 1
+
+            if(Utils.hasPortRange(conf)){
+              0 // pick one form the range
```

Comment:
Maybe it will be clear later on but how returning 0 here will pick the port from range?